karapace/schema_registry_apis.py

```diff
@@ -348,6 +421,26 @@ async def schemas_get(self, content_type, *, schema_id):
             )
         with self.ksr.id_lock:
             schema = self.ksr.schemas.get(schema_id_int)
+
+        def _has_subject_with_id() -> bool:
```

Comment:
Maybe do this before getting `schema`? If this returns `False`, there's no need to get the `schema`, it's set `None` below in that case. Also, maybe use something else than `schema` in the `for schema in val["schemas"].values():` below. Now the same variable name is used for different things in this method, which is slightly confusing.